src/Microsoft.ML.Transforms/GcnTransform.cs

```diff
@@ -225,477 +255,725 @@ public void Save(ModelSaveContext ctx)
+        internal static string TestColumn(ColumnType type)
+        {
+            if ((type.IsVector && !type.IsKnownSizeVector && (type.AsVector.Dimensions.Length > 1)) || type.ItemType != NumberType.R4)
+                return "Expected float or float vector of known size";
+
+            if ((long)type.ValueCount * type.ValueCount > Utils.ArrayMaxSize)
+                return "Vector size exceeds limit";
+
+            return null;
         }
 
         /// <summary>
-        /// Public constructor corresponding to SignatureDataTransform.
+        /// Create a <see cref="LpNormalizingTransform"/> that takes multiple pairs of columns.
         /// </summary>
-        public LpNormNormalizerTransform(IHostEnvironment env, GcnArguments args, IDataView input)
-            : base(env, RegistrationName, env.CheckRef(args, nameof(args)).Column,
-                input, TestIsFloatVector)
+        public LpNormalizingTransform(IHostEnvironment env, params ColumnInfoBase[] columns) :
+           base(Contracts.CheckRef(env, nameof(env)).Register(nameof(LpNormalizingTransform)), GetColumnPairs(columns))
         {
-            Host.AssertNonEmpty(Infos);
-            Host.Assert(Infos.Length == Utils.Size(args.Column));
+            _columns = columns.ToArray();
+        }
 
-            _exes = new ColInfoEx[Infos.Length];
-            for (int i = 0; i < _exes.Length; i++)
-                _exes[i] = new ColInfoEx(args.Column[i], args);
+        // Factory method for SignatureDataTransform for GcnArguments class/>
+        public static IDataTransform Create(IHostEnvironment env, GcnArguments args, IDataView input)
+        {
+            Contracts.CheckValue(env, nameof(env));
+            env.CheckValue(args, nameof(args));
+            env.CheckValue(input, nameof(input));
 
-            // REVIEW: for now check only global (default) values. Move to Bindings/ColInfoEx?
-            if (!args.SubMean && args.UseStdDev)
+            env.CheckValue(args.Column, nameof(args.Column));
+            var cols = new GcnColumnInfo[args.Column.Length];
+            using (var ch = env.Start("ValidateArgs"))
             {
-                using (var ch = Host.Start("Argument validation"))
+                for (int i = 0; i < cols.Length; i++)
                 {
-                    ch.Warning("subMean parameter is false while useStd is true. It is advisable to set subMean to true in case useStd is set to true.");
+                    var item = args.Column[i];
+                    cols[i] = new GcnColumnInfo(item.Source ?? item.Name,
+                        item.Name,
+                        item.SubMean ?? args.SubMean,
+                        item.UseStdDev ?? args.UseStdDev,
+                        item.Scale ?? args.Scale);
                 }
+                if (!args.SubMean && args.UseStdDev)
+                    ch.Warning("subMean parameter is false while useStd is true. It is advisable to set subMean to true in case useStd is set to true.");
```

Comment:
>subMean parameter is false while useStd is true. It is advisable to set subMean to true in case useStd is set to true [](start = 32, length = 117)

this is one creepy warning.